### DISCUSSION TITLE
GPU/Graphics and Rendering Issues for STP Platform (Running sims etc.)

### DISCUSSION THREAD

**Discussion by rcghpge** [SimDynamX/SpaceTeamsPRO#49] [2024 Lunar SAR Competition Q&A] [answered]
I just wanted to add a program issue for further development of the platform. When I open the program, apparently there is a GPU/graphics issue. This is on my desktop environment. ![image](https://github.com/user-attachments/assets/84d4db30-d238-4953-878c-80315cdc6f92)

**Comment by connorjak** [comment, marked as answer]
This appears to be using your CPU graphics, while Space Teams typically requires a dedicated graphics card. See if you can switch to using your dedicated GPU for SpaceCRAFT.exe.

Also, the driver on your CPU graphics will need to be updated, especially if you're on a laptop which typically makes use of both the CPU and GPU graphics.

**Reply by rcghpge** [reply]
This is on my desktop machine. My laptop setup is still having the caching login issue. I went to update the CPU driver but Intel says there are no updates available. I've been running STP w/no issues so far 🙏

**Reply by rcghpge** [reply]
I also have questions. How does the program utilize CPU and GPU resources on individual machines? I assume it runs the GPU on your machine a lot because of the simulation graphics? Knowing what I know about Epic Games tech stack.

**Reply by connorjak** [reply]
Yes, the SpaceCRAFT executable is an Unreal Engine 5 based application that we have programmed to join the simulation and provide all of the graphics and user interaction for the sim. The core of the sim is in the Compute_Server process and various Python System processes that also join the sim as independent Python scripts, using the `spaceteams` python library that we include with STPRO. Read more here: https://github.com/SimDynamX/SpaceTeamsPro/wiki/Sim-Structure

**Comment by rcghpge** [comment]
I figured I aggregate all of my graphics and STP sim rendering issues into 1 post thread. On my laptop environment STP is crashing with a 10061 error code. I couldn't find any other posts in Github Discussions, but it says something about Compute_Server crashing locally or something. I moseyed on over to the crash logs. Just intial evals of the 10061 bug. This is the newest release 0.27.4 btw
![Screenshot (3)](https://github.com/user-attachments/assets/5088c2af-9572-452c-812a-27b789885230)

**Reply by connorjak** [reply]
This is not a graphics / rendering issue. Here is the section of the competition docs for handling errors and crashes: https://github.com/SimDynamX/SpaceTeamsPro/wiki/STU-Lunar-SAR-API-Docs#handling-errors-and-crashes

**Comment by rcghpge** [comment]
Hey Connor, I believe I got a hold of the STP platform now. If you want to close this 1 out. Thanks a ton for the help and pointers
